packages/merge-styles/src/styleToClassName.ts

```diff
@@ -252,17 +252,7 @@ export function applyRegistration(registration: IRegistration, classMap?: { [key
         let selector = rulesToInsert[i];
 
         // Fix selector using map.
-        selector = selector.replace(
-          /(&)|\$([\w-]+)\b/g,
-          (match: string, amp: string, cn: string): string => {
-            if (amp) {
-              return '.' + registration.className;
-            } else if (cn) {
-              return '.' + ((classMap && classMap[cn]) || cn);
-            }
-            return '';
-          }
-        );
+        selector = selector.replace(/(&)/g, '.' + registration.className);
```

Comment:
Technically any sort of behavior change could possibly be considered a breaking change. But I think given that the $ support doesn't really work beyond the initial paint, which has created buggy situations, I'm ok with removing this now. It was long ago removed from the docs. So I think it is wise to just get rid of the overhead here and call it a minor.

Reply:
We can certainly send a mail out about it.

Reply:
nit: Can be `selector.replace(/&/g, ...` as the capture group isn't used. Minor detail, but might help prevent accidental complexity if/when this grows in the future.

Reply:
Also guessing getting rid of the capturing group could marginally increase efficiency?

Reply:
Is there any way we could keep this in for the short term and warn about it? It would really help users understand why specific style files stopped working. Also due to the complex way that some selectors are done, finding files impacted by this could be frustrating.

Reply:
@joschect That's not a bad idea. Maybe we should at least merge a debug assert flag for a couple weeks.